vnext/Shared/OInstance.cpp

```diff
@@ -458,13 +479,35 @@ void InstanceImpl::loadBundleInternal(std::string &&jsBundleRelativePath, bool s
           m_devSettings->platformName,
           /*dev*/ true,
           /*hot*/ false,
-          m_devSettings->inlineSourceMap);
-
-      // Remote debug executor loads script from a Uri, rather than taking the actual bundle string
-      m_innerInstance->loadScriptFromString(
-          std::make_unique<const JSBigStdString>(m_devSettings->useWebDebugger ? bundleUrl : jsBundleString),
-          bundleUrl,
-          synchronously);
+          m_devSettings->inlineSourceMap,
+          hermesBytecodeVersion);
+
+      // This code is based on the HBC Bundle integration on Android
+      // Ref:
+      // https://github.com/facebook/react-native/blob/24d91268b64c7abbd4b26547ffcc663dc90ec5e7/ReactAndroid/src/main/jni/react/jni/CatalystInstanceImpl.cpp#L231
+      if (isHBCBundle(jsBundleString.c_str())) {
+        auto script = std::make_unique<JSBigStdString>(jsBundleString, false);
+        const char *buffer = script->c_str();
+        uint32_t bufferLength = (uint32_t)script->size();
+        uint32_t offset = 8;
+        while (offset < bufferLength) {
+          uint32_t segment = offset + 4;
+          uint32_t moduleLength = bufferLength < segment ? 0 : *(((uint32_t *)buffer) + offset / 4);
```

Comment:
nit: reformat and add parens around this to reduce manual precedeense ordering, there are many ternary,binary and unary operatores in this line:
```
uint32_t moduleLength = bufferLength < segment 
    ? 0 
    : *(((uint32_t *)buffer) + (offset / 4));
```